src/components/WeekDayPicker/index.js

```diff
@@ -0,0 +1,144 @@
+        bottomHelpText,
+        availableDates,
+        disabled,
+        required,
+        readOnly,
+        multiple,
+        error,
+        onChange,
+        className,
+        style,
+    } = props;
+
+    const defaultFieldsetName = useUniqueIdentifier('week-day-items');
+    const fieldsetName = name || defaultFieldsetName;
+
+    const prepareReturnValue = (weekDayValue, isChecked) => {
+        if (multiple) {
+            if (isChecked && !value.includes(weekDayValue)) {
+                return sortWeekDays([...value, weekDayValue]);
+            }
+
+            return sortWeekDays(value.filter(day => day !== weekDayValue));
+        }
+
+        return weekDayValue;
+    };
+
+    const handleOnChange = e => {
+        const weekDayValue = e.currentTarget.value;
+        const isChecked = e.currentTarget.checked;
+
+        return onChange(prepareReturnValue(weekDayValue, isChecked));
+    };
+
+    return (
+        <StyledFieldset className={className} style={style} id={id}>
+            <Label label={label} hideLabel={!label} required={required} readOnly={readOnly} />
+            <WeekDayItems
+                name={fieldsetName}
+                value={value}
+                availableDates={availableDates}
+                disabled={disabled}
+                required={required}
+                readOnly={readOnly}
+                multiple={multiple}
+                error={error}
+                onChange={handleOnChange}
+            />
+            <RenderIf isTrue={!!bottomHelpText}>
+                <StyledHelpText>{bottomHelpText}</StyledHelpText>
+            </RenderIf>
+            <RenderIf isTrue={!!error}>
+                <StyledTextError>{error}</StyledTextError>
+            </RenderIf>
+        </StyledFieldset>
+    );
+}
+
+WeekDayPicker.propTypes = {
+    /** The id of the outer element. */
+    id: PropTypes.string,
+    /** An identifier for the group of radio items. */
+    name: PropTypes.string,
+    /** The value of WeekDayPicker, a list of days selected. */
+    value: PropTypes.oneOfType([
+        PropTypes.arrayOf((propValue, key, componentName, location, propFullName) => {
+            return valuePropValidation(propValue, key, componentName, location, propFullName);
+        }),
+        PropTypes.oneOf([
+            'monday',
+            'tuesday',
+            'wednesday',
+            'thursday',
+            'friday',
+            'saturday',
+            'sunday',
+            '',
+        ]),
+    ]),
+    /** The WeekDayPicker label. */
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    /** Shows the help message below the WeekDayPicker */
+    bottomHelpText: PropTypes.string,
+    /** Specifies the available days from the week for selection. */
+    availableDates: PropTypes.arrayOf((propValue, key, componentName, location, propFullName) => {
+        return valuePropValidation(propValue, key, componentName, location, propFullName);
+    }),
```

Comment:
here I think you can use:
```availableDates: PropTypes.arrayOf(['monday', ...])```

Reply:
Another issue here, I tried and search a bit about it, I reviewed another arrayOf examples used inside the project and I didn't found any clues on how to implement.

Also I reviewed Google and arrayOf could be only for PropTypes types (string, number, etc) I might be wrong.

The only way I saw was implementing the same method above used.

If you see the "value PropTypes configuration", you will see this same code to be able to have an arraOf specific items.

What I am seeing here is that maybe we could extend this method to one of ours to be able to pass an extra parameter to valuePropValidation and maybe we can create a PropTypes helper function for custom arrayOf specific items.

Maybe I am bit crazy I don't know jahahahaha se me ocurrió, I saw an oportunity to extend this prop type method configuration for arrayOf